test/libyul/objectCompiler/subObjectAccess.yul

```diff
@@ -0,0 +1,41 @@
+object "A" {
+  code {
+    pop(dataoffset("A"))
+    pop(dataoffset("B"))
+    pop(dataoffset("A.B"))
```

Comment:
Is this actually what we wanted to implement? At least as far as shown in this example, there is no benefit in using this dot-syntax. Don't we need access (from here) to `B.C`?

Reply:
I agree with you. I just copied the same example from original issue.

Reply:
So, accessing `B.C` isn't possible with these changes. I am not completely sure if this is possible at all? Look at this example:

```
object "A" {
  code {
    pop(dataoffset("A"))
    pop(dataoffset("B"))
    pop(dataoffset("B.C"))   // doesn't work
  }

  object "subobject" {
    code {
      pop(dataoffset("C"))
    }
    object "subsubobject" {
      code {
        invalid()
      }
    }
  }
}
```

Not sure what would be the way to access sub objects of subobjects? `dataoffset` and other builtins are searching for objects only in direct subobjects. @chriseth what if we add `subsubobject` as sub-object to `A` in assembly?

Reply:
The use-case is implementing `type(C).runtimeCode` for a different contract `C`.

Reply:
I am trying to solve this #6739

Was thinking about something like this. So, `B.C` would be `sub_1` in this case 
```
object "A" {
  code {
    pop(dataoffset("B.C"))
  }

  object "B" {
    code {
      pop(dataoffset("C"))
    }
    object "C" {
      code {
        invalid()
      }
    }
  }
}
// ----
// Assembly:
//   pop(dataOffset(sub_1))
// stop
//
// sub_0: assembly {
//     pop(dataOffset(sub_0))
//     stop
//
//     sub_0: assembly {
//         invalid
//     }
// }
// sub_1: assembly {    // this is added as we needed to acces `B.C` from `A`
//     invalid
// }
```

for `type(C).runtimeCode` we need access to runtime object of `C`
Question was, should the object that needs runtimeCode include runtime object as subobject?

Reply:
It would be good if we would not have to copy the object. Can you check how it was done in the old code generator?

Reply:
I will check now

Reply:
```
contract D {
    constructor() public {}
}
contract C {
    function test() public returns (bytes32) {
	    return keccak256(type(D).runtimeCode);
    }
}
```
In this example in runtime part of assembly for object C it adds sub object which is actually the one that is exactly the same like runtime object of D. 
So it does copy the whole object...

But, maybe we can come up with something new for yul? I do not have an idea really, but am willing to discuss possibilities :-)